src/Blazored.LocalStorage/LocalStorageService.cs

```diff
@@ -15,12 +15,21 @@ public LocalStorageService(IJSRuntime jSRuntime)
             _jSInProcessRuntime = jSRuntime as IJSInProcessRuntime;
         }
 
-        public Task SetItem(string key, object data)
+        public async Task SetItem(string key, object data)
         {
             if (string.IsNullOrEmpty(key))
                 throw new ArgumentNullException(nameof(key));
 
-            return _jSRuntime.InvokeAsync<object>("Blazored.LocalStorage.SetItem", key, Json.Serialize(data));
+            RaiseOnChanging(key, data, out ChangingEventArgs e);
+
+            if (e.Cancel)
+                return;
+
+            _ = await _jSRuntime.InvokeAsync<object>("Blazored.LocalStorage.SetItem", key, Json.Serialize(data));

```

Comment:
There is no need to use a discard here. You can just await the call.

Reply:
(The IDE suggested it, because you were previously returning the value.)